plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/ui/LoginServiceUi.java

```diff
@@ -0,0 +1,85 @@
+/*******************************************************************************
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License v1.0 which
+ * accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package com.google.cloud.tools.eclipse.appengine.login.ui;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeRequestUrl;
+import com.google.cloud.tools.eclipse.appengine.login.Messages;
+import com.google.cloud.tools.ide.login.UiFacade;
+import com.google.cloud.tools.ide.login.VerificationCodeHolder;
+
+import org.eclipse.jface.dialogs.InputDialog;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.swt.program.Program;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.ui.IWorkbench;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.commands.ICommandService;
+
+public class LoginServiceUi implements UiFacade {
+
+  public static void showErrorDialogHelper(String title, String message) {
+    Shell shell = PlatformUI.getWorkbench().getDisplay().getActiveShell();
+    MessageDialog.openError(shell, title, message);
+  }
+
+  @Override
+  public boolean askYesOrNo(String title, String message) {
+    throw new RuntimeException("Not allowed to ensure non-UI threads don't prompt."); //$NON-NLS-1$
+  }
+
+  @Override
+  public void showErrorDialog(String title, String message) {
```

Comment:
We may enforce the policy that login is done only through explicit user selection of a button or menu. Currently, the deployment automatically triggers login rather than simply failing with a message saying that the user needs to login first. Here is the relevant issue: #472. We could make a decision soon.